### ISSUE TITLE
SMT crashes on increment assignment operator within if statemntes

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#4537]
In https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/ECRecovery.sol

```
    if (v < 27) {
       v += 27;
    }
```

causes a crash

```
/Users/alex/Projects/solidity/libsolidity/formal/SMTChecker.cpp(757): Throw in function void dev::solidity::SMTChecker::mergeVariables(const vector<const dev::solidity::VariableDeclaration *> &, const smt::Expression &, const dev::solidity::SMTChecker::VariableSequenceCounters &, const dev::solidity::SMTChecker::VariableSequenceCounters &)
Dynamic exception type: boost::exception_detail::clone_impl<dev::solidity::InternalCompilerError>
std::exception::what: 
[dev::tag_comment*] = 
```

Changing it to `v = v + 27;` "fixes" it.

Also a separate issue: no warnings are displayed for unsupported assembly block and as a result it just complains:
```
ECRecovery.sol:45:9: Warning: Condition is always true.
    if (v < 27) {
        ^----^
```

**Comment by leonardoalt** [comment]
Missing support to compound assignments

**Comment by leonardoalt** [comment]
Fixed by #6006 
Created #6418 to track unsupported assembly blocks

**Comment by fty48jj19** [comment]
ok
